llvm/lib/IR/VectorTypeUtils.cpp

```diff
@@ -22,7 +34,11 @@ Type *llvm::toVectorizedStructTy(StructType *StructTy, ElementCount EC) {
          "expected all element types to be valid vector element types");
   return StructType::get(
       StructTy->getContext(),
-      map_to_vector(StructTy->elements(), [&](Type *ElTy) -> Type * {
+      map_to_vector(enumerate(StructTy->elements()), [&](auto It) -> Type * {
+        Type *ElTy = It.value();
+        if (IID != 0 &&
+            isVectorIntrinsicWithStructReturnScalarAtField(IID, It.index()))
```

Comment:
```suggestion
        if (
            isVectorIntrinsicWithStructReturnScalarAtField(IID, It.index()))
```

IID ==0 should naturally be handled by `isVectorIntrinsicWithStructReturnScalarAtField`